securecookie.go

```diff
@@ -319,6 +348,26 @@ func deserialize(src []byte, dst interface{}) error {
 	return nil
 }
 
+// serialize encodes a value using encoding/json.
+func (e JSONEncoder) serialize(src interface{}) ([]byte, error) {
+	buf := new(bytes.Buffer)
+	enc := json.NewEncoder(buf)
+	if err := enc.Encode(src); err != nil {
+		return nil, err
+	}
+
+	return buf.Bytes(), nil
+}
+
+// deserialize decodes a value using encoding/json.
+func (e JSONEncoder) deserialize(src []byte, dst interface{}) error {
+	dec := json.NewDecoder(bytes.NewBuffer(src))
```

Comment:
This can be `NewReader`, but I guess you just copied that from the Gob version so it's not your fault ;)